backend/src/main/java/de/amos/apachepulsarui/dto/TopicDto.java

```diff
@@ -5,28 +5,48 @@
 
 package de.amos.apachepulsarui.dto;
 
-import lombok.AccessLevel;
+import lombok.Builder;
 import lombok.Data;
-import lombok.NoArgsConstructor;
 import org.apache.pulsar.common.naming.TopicName;
+import org.apache.pulsar.common.policies.data.PublisherStats;
+import org.apache.pulsar.common.policies.data.TopicStats;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
 
 @Data
-@NoArgsConstructor(access = AccessLevel.PRIVATE)
+@Builder
 public class TopicDto {
 
     private String name;
 
-	private String namespace;
+    private String namespace;
+
+    private String tenant;
+
+    private Set<String> subscriptions;
 
-	private String tenant;
+    private List<String> producers;
+
+    public static TopicDto create(String completeTopicName, TopicStats topicStats) {
+        Set<String> subscriptions = topicStats.getSubscriptions().keySet();
```

Comment:
Is it important to have a list there? I don't have a problem with that, I was just wondering because we otherwise always use lists : )

Reply:
That because of the  hashmap which contains the subscriptions with its data.